qiskit_nature/operators/second_quantization/qubit_converter.py

```diff
@@ -40,58 +106,235 @@ def z2symmetry_reduction(self) -> Optional[List[int]]:
+
+        Returns:
+            A qubit operator or list of thereof the same length as the second_q_ops list. All
+            operators in the second_q_ops list must commute with the symmetry detected when
+            :meth:`convert` was called. If it does not then the position in the output list
+            will be set to `None` to preserve the order; or None may be directly returned
+            in the case when a single operator is provided
+
+        Raises:
+            QiskitNatureError: If property is accessed before a successful conversion
+        """
+        self._check_valid(True)
+
+        # To allow a single operator to be converted, but use the same logic that does the
+        # actual conversions, we make a single entry list of it here and unwrap to return.
+        wrapped = False
+        if isinstance(second_q_ops, SecondQuantizedOp):
+            second_q_ops = [second_q_ops]
+            wrapped = True
+
+        qubit_ops = self._map_to_qubits(second_q_ops)
+        qubit_ops_reduced, _ = self._two_qubit_reduce(qubit_ops, self.num_particles)
+        qubit_ops_tapered = self._symmetry_reduce(qubit_ops_reduced)
+
+        if wrapped:
+            qubit_ops_tapered = qubit_ops_tapered[0]
+
+        return qubit_ops_tapered
 
     def _map_to_qubits(self, second_q_ops: List[SecondQuantizedOp]) -> List[PauliSumOp]:
-        if isinstance(self._mappers, tuple):
-            raise NotImplementedError
 
         qubit_ops: List[PauliSumOp] = []
         for op in second_q_ops:
-            qubit_ops.append(self._mappers.map(op))
+            qubit_ops.append(self._mapper.map(op))
 
         return qubit_ops
 
-    def _symmetry_reduce(self, qubit_ops: List[PauliSumOp]) -> List[Optional[PauliSumOp]]:
+    def _two_qubit_reduce(self, qubit_ops: List[PauliSumOp],
+                          num_particles: Optional[Tuple[int, int]])\
+            -> Tuple[List[Optional[PauliSumOp]], bool]:
+        if self._two_qubit_reduction and not self._mapper.allows_two_qubit_reduction:
+            logger.warning("Ignoring requested two qubit reduction as mapping does not support it")
 
-        if self.z2symmetry_reduction is None:
+        did_2_q = False
+        if (self._two_qubit_reduction and
+                self._mapper.allows_two_qubit_reduction and
+                num_particles is not None):
+            two_q_reducer = TwoQubitReduction(num_particles)
+            reduced_qubit_ops = [cast(PauliSumOp, two_q_reducer.convert(op)) for op in qubit_ops]
+            did_2_q = True
+        else:
             reduced_qubit_ops = qubit_ops
-            self._z2symmetries = Z2Symmetries([], [], [], None)
+
+        return reduced_qubit_ops, did_2_q
+
+    def _find_symmetry_reduce(self, qubit_ops: List[PauliSumOp])\
+            -> Tuple[List[Optional[PauliSumOp]], Z2Symmetries]:
+
+        if self.z2symmetry_reduction is None:
+            tapered_qubit_ops = qubit_ops
+            z2_symmetries = self._no_symmetries
```

Comment:
If you add a return statement here, you can get red if of `else` and dedent the rest of this function.

Reply:
I prefer a single return at the end of a method unless its an exception out of it.